packages/binding-file/src/file-client.ts

```diff
@@ -17,59 +17,43 @@
  * File protocol binding
  */
 import { Form, SecurityScheme } from "@node-wot/td-tools";
-import { ProtocolClient, Content, createLoggers } from "@node-wot/core";
+import { ProtocolClient, Content, createLoggers, ContentSerdes } from "@node-wot/core";
 import { Subscription } from "rxjs/Subscription";
 import fs = require("fs");
-import path = require("path");
+import { fileURLToPath } from "node:url";
 
-const { debug, warn } = createLoggers("binding-file", "file-client");
-
-/**
- * Used to determine the Content-Type of a file from the extension in its
- * {@link filePath} if no explicit Content-Type is defined.
- *
- * @param filepath The file path the Content-Type is determined for.
- * @returns An appropriate Content-Type or `application/octet-stream` as a fallback.
- */
-function mapFileExtensionToContentType(filepath: string) {
-    const fileExtension = path.extname(filepath);
-    debug(`FileClient found '${fileExtension}' extension`);
-    switch (fileExtension) {
-        case ".txt":
-        case ".log":
-        case ".ini":
-        case ".cfg":
-            return "text/plain";
-        case ".json":
-            return "application/json";
-        case ".jsontd":
-            return "application/td+json";
-        case ".jsonld":
-            return "application/ld+json";
-        default:
-            warn(`FileClient cannot determine media type for path '${filepath}'`);
-            return "application/octet-stream";
-    }
-}
+const { debug } = createLoggers("binding-file", "file-client");
 
 export default class FileClient implements ProtocolClient {
     public toString(): string {
         return "[FileClient]";
     }
 
-    private async readFile(filepath: string, contentType?: string): Promise<Content> {
-        const resource = fs.createReadStream(filepath);
-        const resourceContentType = contentType ?? mapFileExtensionToContentType(filepath);
-        return new Content(resourceContentType, resource);
+    private async readFromFile(uri: string, contentType: string) {
+        const filePath = fileURLToPath(uri);
+        debug(`Reading file of Content-Type ${contentType} from path ${filePath}.`);
+
+        const resource = fs.createReadStream(filePath);
+        return new Content(contentType, resource);
     }
 
     public async readResource(form: Form): Promise<Content> {
-        const filepath = new URL(form.href).pathname;
-        return this.readFile(filepath, form.contentType);
+        const formContentType = form.contentType;
+        if (form.contentType == null) {
```

Comment:
```suggestion
        if (formContentType == null) {
```

I know it is the same but since you have the variable already it makes more sense to me

Reply:
Thank you, good catch :)